extension/encoding/awscloudwatchmetricstreamsencodingextension/opentelemetry_unmarshaler_test.go

```diff
@@ -3,8 +3,74 @@
 
 package awscloudwatchmetricstreamsencodingextension
 
-import "testing"
+import (
+	"errors"
+	"testing"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/golden"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest/pmetrictest"
+)
+
+// getRecordFromFile reads the pmetric.Metrics
+// from filename and returns the record in the
+// format the encoding extension expects the
+// metrics to be
+func getRecordFromFile(t *testing.T, filename string) []byte {
+	metrics, err := golden.ReadMetrics(filename)
+	require.NoError(t, err)
+
+	m := pmetric.ProtoMarshaler{}
+	data, err := m.MarshalMetrics(metrics)
+	require.NoError(t, err)
+
+	record := proto.EncodeVarint(uint64(len(data)))
+	record = append(record, data...)
+	return record
+}
 
 func TestUnmarshalOpenTelemetryMetrics(t *testing.T) {
-	t.Skip("still to be implemented")
+	t.Parallel()
+
+	unmarshaler := formatOpenTelemetry10Unmarshaler{}
+	tests := map[string]struct {
+		record                  []byte
+		expectedMetricsFilename string
+		expectedErr             error
+	}{
+		"valid_record": {
+			record:                  getRecordFromFile(t, "testdata/opentelemetry1/valid_metric.yaml"),
+			expectedMetricsFilename: "testdata/opentelemetry1/valid_metric_expected.yaml",
+		},
```

Comment:
Could you please add another case where we concatenate two of these?

Reply:
Yes. There was a small error, and metrics for same resource were giving a new resource. I have changed it to use the `expmetrics.Merge` instead now.

Reply:
> Yes. There was a small error, and metrics for same resource were giving a new resource. I have changed it to use the expmetrics.Merge instead now.

Let's stick to the original behaviour please: if a resource is duplicated in the input, then they can be duplicated in the output.

> But this means it will have different behavior now from the unmarshaler of awsfirehosereceiver. But since we do the same for the JSON unmarshaler, it seems to make sense to keep the consistency in this extension.

The JSON format is quite different though: there's no way to have a resource hierarchy there, so the resource attributes are added to each metric.